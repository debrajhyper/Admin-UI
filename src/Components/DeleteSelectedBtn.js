import React from 'react'

import { FaTrash } from 'react-icons/fa';

const DeleteSelectedBtn = ({ users, setUsers }) => {

    //  Delete the selected users from the users array and set the users to the state using the setUsers() method.
    const handleDeleteSelected = e => {
        e.preventDefault();
        let newUsers = [...users];
        newUsers = newUsers.filter(user => {
            return !user.isChecked;
        });
        setUsers(newUsers);
    };
    
    return (
        <button 
            data-testid="delete-selected-btn"
            id="first" 
            type="button"
            title="Delete selected"
            className="flex items-center justify-center p-2 md:px-4 shadow-md shadow-red-300 dark:shadow-red-900 text-white bg-red-500 transition-all duration-500 rounded focus:shadow hover:bg-red-600 hover:shadow-lg hover:shadow-red-200 outline-none focus:outline-none" 
            onClick={handleDeleteSelected} 
        >
            <FaTrash />
            <span className="ml-1 md:ml-2 text-xs md:text-base text-left">
                Delete Selected
            </span>
        </button>
    )
}

export default DeleteSelectedBtn
