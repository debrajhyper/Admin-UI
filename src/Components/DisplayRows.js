import React from 'react'

import { FaUserEdit, FaTrash } from 'react-icons/fa'

const DisplayRows = ({ user, users, setUsers, setEditUserId, setEditUserForm }) => {

    //  Check if the user is checked and set the users to the state using the setUsers() method.
    const handleChecked = e => {
        const { id, value, checked } = e.target;
        let newUsers = users.map(user => user.id === id && user.name === value ? { ...user, isChecked: checked } : user);
        setUsers(newUsers);
    };

    //  Edit the user data and set the user data to the state using the setEditUserForm() method.
    const handleUserEdit = (e, user) => {
        e.preventDefault();
        setEditUserId(user.id);
        const formValues = {
            id: user.id,
            name: user.name,
            email: user.email,
            role: user.role
        };
        setEditUserForm(formValues);
    };
    
    //  Delete the user from the users array and set the users to the state using the setUsers() method.
    const handleUserDelete = userId => {
        const newUsers = [ ...users ];
        const index = users.findIndex(user => user.id === userId);
        newUsers.splice(index, 1);
        setUsers(newUsers);
    };

    return (
        <tr id={user.id} className={`data-row bg-white whitespace-nowrap hover:cursor-default hover:bg-blue-light hover:text-white ${user?.isChecked ? 'bg-gray-100' : null}`}>
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <input 
                    type="checkbox" 
                    className="form-checkbox h-4 w-4 outline-none focus:outline-none" 
                    id={user.id}
                    onChange={handleChecked}
                    checked={user?.isChecked || false}
                    value={user.name}
                />
            </td>
            <td className="px-6 py-2 whitespace-nowrap">{user.name}</td>
            <td className="px-6 py-2 whitespace-nowrap">{user.email}</td>
            <td className="px-6 py-2 whitespace-nowrap">{user.role}</td>
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <button 
                    type="button" 
                    title="Edit" 
                    className="btn p-2 mr-2 text-blue-light rounded" 
                    onClick={(e) => handleUserEdit(e, user)} 
                >
                    <FaUserEdit size={16} />
                </button>
                <button 
                    type="button" 
                    title="Delete" 
                    className="btn p-2 ml-2 text-red-500 rounded" 
                    onClick={() => handleUserDelete(user.id)} 
                >
                    <FaTrash size={16} />
                </button>
            </td>
        </tr>
    )
}

export default DisplayRows
