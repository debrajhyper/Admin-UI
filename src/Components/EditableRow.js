import React from 'react';

import { FiCheckCircle } from 'react-icons/fi';
import { MdOutlineCancel } from 'react-icons/md';

const EditableRow = ({ user, editUserForm ,setEditUserForm, setEditUserId }) => {

    //  When user change any value of the edited user then update the editUserForm state.
    const handleUserEditFormChange = e => {
        e.preventDefault();
        const { name, value } = e.target;
        setEditUserForm({
            ...editUserForm,
            [name]: value
        });
    };
    
    //  Cancel the edit of the user and set the editUserForm state to an empty object.
    const handleUserEditFormCancel = e => {
        e.preventDefault();
        setEditUserId(null);
    };

    return (
        <tr id={user.id} className="data-row bg-white dark:bg-dark whitespace-nowrap hover:cursor-default hover:bg-blue-light dark:hover:bg-blue-dark hover:text-gray">
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <input 
                type="checkbox" 
                className="form-checkbox h-4 w-4 outline-none focus:outline-none" 
                id={user.id}
                // checked={user?.isChecked || false}
                value={user.name}
                disabled={true}
                />
            </td>
            <td className="px-5 py-2 whitespace-nowrap text-left">
                <input 
                type="text" 
                name="name" 
                className="form-input px-1 border-2 border-blue-light dark:border-blue-dark rounded bg-indigo-100 dark:bg-indigo-400 focus:outline-none" 
                required="required" 
                placeholder="Enter a name"
                value={user.name}
                onChange={handleUserEditFormChange}
                size={14}
                />
            </td>
            <td className="px-5 py-2 whitespace-nowrap text-left">
                <input 
                type="text" 
                name="email"
                className="form-input px-1 border-2 border-blue-light dark:border-blue-dark rounded bg-indigo-100 dark:bg-indigo-400 focus:outline-none" 
                required="required" 
                placeholder="Enter an email"
                value={user.email}
                onChange={handleUserEditFormChange}
                size={25}
                />
            </td>
            <td className="px-5 py-2 whitespace-nowrap text-left">
                <input 
                type="text" 
                name="role"
                className="form-input px-1 border-2 border-blue-light dark:border-blue-dark rounded bg-indigo-100 dark:bg-indigo-400 focus:outline-none" 
                required="required"
                placeholder="Enter a role"
                value={user.role}
                onChange={handleUserEditFormChange}
                size={9}
                />
            </td>
            <td className="px-6 py-2 whitespace-nowrap text-center">
                <button type="submit" title="Update" className="btn p-2 mr-2 text-green-500 rounded">
                    <FiCheckCircle size={16} />
                </button>
                <button type="button" title="Cancel" className="btn p-2 ml-2 text-red-500 rounded" onClick={handleUserEditFormCancel}>
                    <MdOutlineCancel size={16} />
                </button>
            </td>
        </tr>
    )
}

export default EditableRow
