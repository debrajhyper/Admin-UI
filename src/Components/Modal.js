import React from 'react'
import icon from "../Images/1140-error-outline.webp";

const Modal = () => {
    return (
        <div className="bg-slate-800 bg-opacity-50 flex justify-center items-center z-50 absolute top-0 right-0 bottom-0 left-0">
            <div className="bg-red-100 text-red-600 border border-red-200 px-2 md:px-16 py-6 md:py-10 rounded-md text-center" role="alert">
                <div className="flex justify-center items-center mb-4">
                    <img src={icon} alt="error" className="w-20" />
                </div>
                <h1 className="text-xl mb-4 font-bold">Failed to load Data from Server</h1>
                <p className="text-sm">Please check your internet connection and try again</p>
            </div>
        </div>
    )
}

export default Modal