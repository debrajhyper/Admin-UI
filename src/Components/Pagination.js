import React from 'react'

import { HiChevronDoubleLeft, HiChevronDoubleRight } from 'react-icons/hi';
import { IoIosArrowBack, IoIosArrowForward } from 'react-icons/io';

const Pagination = ({ users, itemPerPage, currentPage,  setCurrentPage }) => {

    //  Array of numbers containing the number of pages.
    var pages = [];

    //  Calculate the number of pages.
    for(let i=1; i<=Math.ceil(users.length/itemPerPage); i++) {
        pages.push(i);
    };

    //  set the current page to the clicked page number.
    const handleClick = e => {
        setCurrentPage(Number(e.target.id));
    };

    //  paginate to the first page.
    const handleFirstBtn = () => {
        setCurrentPage(1);
    };

    //  paginate to the previous page.
    const handlePrevBtn = () => {
        setCurrentPage(currentPage - 1);
    };

    //  page numbers to be displayed in the pagination.
    const renderPageNumbers = pages.map(number => {
        return (
            <li key={number}>
                <button 
                    id={number} 
                    className={`${currentPage === number ? "active" : null} w-8 md:w-10 h-8 md:h-10 bg-white text-indigo-600 transition-colors duration-100 rounded-full focus:shadow-outline hover:bg-indigo-100`} 
                    onClick={handleClick}
                >
                    {number}
                </button>
            </li>
        );
    });

    //  paginate to the next page.
    const handleNextBtn = () => {
        setCurrentPage(currentPage + 1);
    };

    //  paginate to the last page.
    const handleLastBtn = () => {
        setCurrentPage(Math.ceil(users.length/itemPerPage));
    };

    return (
        <nav id="pagination" aria-label="Page navigation">
            <ul className="inline-flex space-x-1 sm:space-x-2 md:space-x-4 px-auto text-center">
                <li>
                    <button 
                    id="first" 
                    className="flex items-center justify-center w-8 md:w-10 h-8 md:h-10 text-indigo-600 transition-colors duration-150 rounded-full focus:shadow-outline hover:bg-indigo-100 disabled:opacity-60 disabled:bg-transparent" 
                    onClick={handleFirstBtn} 
                    disabled={currentPage === pages[0] ? true : false} 
                    >
                    <HiChevronDoubleLeft />
                    </button>
                </li>
                <li>
                    <button 
                    id="prev" 
                    className="flex items-center justify-center w-8 md:w-10 h-8 md:h-10 text-indigo-600 transition-colors duration-150 rounded-full focus:shadow-outline hover:bg-indigo-100 disabled:opacity-60 disabled:bg-transparent" 
                    onClick={handlePrevBtn} 
                    disabled={currentPage === pages[0] ? true : false}
                    >
                    <IoIosArrowBack />
                    </button>
                </li>
                {renderPageNumbers}
                <li>
                    <button 
                    id="next" 
                    className="flex items-center justify-center w-8 md:w-10 h-8 md:h-10 text-indigo-600 transition-colors duration-150 rounded-full focus:shadow-outline hover:bg-indigo-100 disabled:opacity-60 disabled:bg-transparent" 
                    onClick={handleNextBtn} 
                    disabled={currentPage === pages[pages.length - 1] ? true : false} 
                    >
                    <IoIosArrowForward />
                    </button>
                </li>
                <li>
                    <button 
                    id="last" 
                    className="flex items-center justify-center w-8 md:w-10 h-8 md:h-10 text-indigo-600 transition-colors duration-150 rounded-full focus:shadow-outline hover:bg-indigo-100 disabled:opacity-60 disabled:bg-transparent" 
                    onClick={handleLastBtn} 
                    disabled={currentPage === pages[pages.length - 1] ? true : false}
                    >
                    <HiChevronDoubleRight />
                    </button>
                </li>
            </ul>
        </nav>
    )
}

export default Pagination
