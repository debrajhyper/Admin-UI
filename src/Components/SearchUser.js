import React from 'react'

const SearchUser = ({ searchResult, setSearchResult }) => {

    //  Method to set the search text entered by the user to the state using the setSearchResult() method.
    const searchHandler = event => {
        setSearchResult(event.target.value);
    };

    return (
        <input 
        data-testid="search-user"
        type="text" 
        placeholder="Search"
        className="float-right outline-none border-2 dark:border-gray-700 rounded pl-2 px-1 md:px-2 py-1 dark:bg-gray-800 hover:border-blue-light dark:hover:border-blue-dark focus:border-blue-light dark:focus:border-blue-dark" 
        value={searchResult}
        onChange={searchHandler} 
        />
    )
}

export default SearchUser
