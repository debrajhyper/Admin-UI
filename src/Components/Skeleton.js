import React from 'react'
import icon from '../Images/105-loader.webp'

const Skeleton = () => {
    return (
        <tr className="text-center w-full text-lg text-indigo-500">
            <td colSpan={5} className="p-20">
                <div className="flex justify-center items-center text-3xl">
                    <h1 className="px-1 pt-2 font-bold tracking-widest">Loading</h1>
                    <img src={icon} alt="loader" className="w-20 pb-2"/>
                </div>
            </td>
        </tr>
    )
}

export default Skeleton