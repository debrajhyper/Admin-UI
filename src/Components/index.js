export { default as SearchUser } from './SearchUser';
export { default as EditableRow } from './EditableRow';
export { default as DisplayRows } from './DisplayRows';
export { default as Pagination } from './Pagination';
export { default as DeleteSelectedBtn } from './DeleteSelectedBtn';
export { default as Modal } from './Modal';
export { default as Skeleton } from './Skeleton';