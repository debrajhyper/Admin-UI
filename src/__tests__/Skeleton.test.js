import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Skeleton } from "../Components";

afterEach(cleanup);

test("render without crashing", () => {
    const { container } = render(<Skeleton />);
    expect(container).toBeInTheDocument();
    expect(container).toContainHTML("<tr>");
    expect(container).toContainHTML("<td>");
})

test("snapshot", () => {
    const tree = renderer.create(<Skeleton />).toJSON();
    expect(tree).toMatchSnapshot();
})