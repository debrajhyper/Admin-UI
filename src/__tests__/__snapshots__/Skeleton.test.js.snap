// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<tr
  className="text-center w-full text-lg text-indigo-500"
>
  <td
    className="p-20"
    colSpan={5}
  >
    <div
      className="flex justify-center items-center text-3xl"
    >
      <h1
        className="px-1 pt-2 font-bold tracking-widest"
      >
        Loading
      </h1>
      <img
        alt="loader"
        className="w-20 pb-2"
        src="105-loader.webp"
      />
    </div>
  </td>
</tr>
`;
